tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/quickaccess/QuickAccessProvidersTest.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * @since 3.4
+ */
+@RunWith(JUnit4.class)
+public class QuickAccessProvidersTest {
+	private static final String ACTIVITY_ID = "org.eclipse.ui.tests.activitySupportTest.issue1832";
+	private final String COMMAND_ID = "org.eclipse.ui.tests.activitySupportTest.commands.issue1832";
+	private static final String ACTION_LABEL = "Add Java Exception Breakpoint...";
+
+	@Test
+	public void testCommandProvider() {
+		IWorkbenchActivitySupport workbenchActivitySupport = PlatformUI.getWorkbench().getActivitySupport();
+		CommandProvider cmdProvider = new CommandProvider();
+		IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+		WorkbenchWindow workbenchWindow = (WorkbenchWindow) window;
+		final MWindow model = workbenchWindow.getModel();
+		cmdProvider.setContext(model.getContext().getActiveLeaf());
+
+		QuickAccessElement[] elementsBefore = cmdProvider.getElements();
+		assertCommandAbsent(elementsBefore);
+
+		// enable the test activity id
+		Set<String> enabledActivityIds = workbenchActivitySupport.getActivityManager().getEnabledActivityIds();
+		Set<String> toBeEnabled = new HashSet<>();
+		toBeEnabled.add(ACTIVITY_ID);
+		toBeEnabled.addAll(enabledActivityIds);
+		workbenchActivitySupport.setEnabledActivityIds(toBeEnabled);
+
+		CommandProvider cmdProvider1 = new CommandProvider();
+		cmdProvider1.setContext(model.getContext().getActiveLeaf());
+		QuickAccessElement[] elementsAfter = cmdProvider1.getElements();
+		assertCommandPresent(elementsAfter);
+	}
+
+	@Test
+	public void testActionProvider() {
+		IWorkbenchActivitySupport workbenchActivitySupport = PlatformUI.getWorkbench().getActivitySupport();
+		ActionProvider actionProvider = new ActionProvider();
+		QuickAccessElement[] actionElements = actionProvider.getElements();
+		assertActionPresent(actionElements);
+
+		Set<String> enabledActivityIdsBefore = new HashSet<>();
+		enabledActivityIdsBefore.addAll(workbenchActivitySupport.getActivityManager().getEnabledActivityIds());
+
+		// disable java development. set any new activity id it disables other id's
+		Set<String> toBeEnabled = new HashSet<>();
+		toBeEnabled.add(ACTIVITY_ID);
+		workbenchActivitySupport.setEnabledActivityIds(toBeEnabled);
+
+		ActionProvider actionProvider1 = new ActionProvider();
+		QuickAccessElement[] actionElements1 = actionProvider1.getElements();
+		assertActionAbsent(actionElements1);
+
+		// restore previously enabled activity id's
+		workbenchActivitySupport.setEnabledActivityIds(enabledActivityIdsBefore);
+
+		// make sure restore was successful
+		ActionProvider actionProvider2 = new ActionProvider();
+		QuickAccessElement[] actionElements2 = actionProvider2.getElements();
+		assertActionPresent(actionElements2);
+	}
+
+	private void assertActionAbsent(QuickAccessElement[] actionElements) {
+		boolean present = false;
+		for (QuickAccessElement quickAccessElement : actionElements) {
+			if (quickAccessElement.getLabel().equals(ACTION_LABEL)) {
```

Comment:
nit: to avoid possible NPE use constantString.equals() or symmetric Objects.equals